libs/ui-watt/src/lib/components/text-field/watt-text-field.stories.ts

```diff
@@ -0,0 +1,164 @@
+    label: 'Input',
+    exampleFormControl: new FormControl(null),
+  },
+  template,
+});
+
+Input.parameters = {
+  docs: {
+    source: {
+      code: howToUseGuideBasic,
+    },
+  },
+};
+
+export const WithDisabled: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'disabled',
+    exampleFormControl: new FormControl({ value: null, disabled: true }),
+  },
+  template,
+});
+
+WithDisabled.parameters = {
+  docs: {
+    source: {
+      code: `new FormControl({ value: null, disabled: true })`,
+    },
+  },
+};
+
+export const WithValue: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'withValue',
+    exampleFormControl: new FormControl({ value: 'this is a value', disabled: false }),
+  },
+  template,
+});
+
+WithValue.parameters = {
+  docs: {
+    source: {
+      code: `new FormControl({ value: 'this is a value', disabled: false })`,
+    },
+  },
+};
+
+export const WithPlaceholder: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'withPlaceholder',
+    exampleFormControl: new FormControl(null),
+    placeholder: "I'm a placeholder",
+  },
+  template,
+});
+
+export const WithNumber: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'number',
+    exampleFormControl: new FormControl(null),
+    type: 'number',
+  },
+  template,
+});
+
+export const WithRequired: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'required',
+    exampleFormControl: new FormControl(null),
+  },
+  template: `<watt-text-field [required]="true" [label]="label" [type]="type" [placeholder]="placeholder" [formControl]="exampleFormControl">
+              <span info *ngIf="exampleFormControl.dirty && exampleFormControl.hasError('required')">This field is required</span>
+            </watt-text-field>`,
+});
+
+export const WithPrefix: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'Prefix',
+    exampleFormControl: new FormControl(null),
+  },
+  template: `<watt-text-field prefix="search" [label]="label" [required]="required" [type]="type" [placeholder]="placeholder" [formControl]="exampleFormControl" />`,
+});
+
+export const WithSuffix: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'Suffix',
+    exampleFormControl: new FormControl(null),
+  },
+  template: `<watt-text-field [label]="label" [required]="required" [type]="type" [placeholder]="placeholder" [formControl]="exampleFormControl">
+              <watt-button variant="icon" icon="search" />
+             </watt-text-field>`,
+});
+
+export const WithHint: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'hint',
+    exampleFormControl: new FormControl(null),
+  },
+  template: `<watt-text-field [required]="true" [label]="label" [type]="type" [placeholder]="placeholder" [formControl]="exampleFormControl">
+              <span error>This field is required</span>
+              <span hint>This is a hint</span>
+            </watt-text-field>`,
```

Comment:
issue: there's a small jump when the component goes from normal to error state (when the "hint" gets hidden and "error" is shown).

![issue](https://github.com/Energinet-DataHub/greenforce-frontend/assets/1096332/1065de8e-d4c6-464c-8fbc-3892db2324c9)